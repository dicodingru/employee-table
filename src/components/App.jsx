import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import FilterableEmployeeTable from './FilterableEmployeeTable';
import EmployeeEditForm from './EmployeeEditForm';
import NotFound from './NotFound.jsx';

const App = () => (
  <Router>
    <div className="container-fluid py-3">
      <Switch>
        <Route exact path="/" component={FilterableEmployeeTable} />
        <Route
          exact
          path="/edit/:id"
          render={({ location }) => {
            const employeeId = location.pathname.replace('/edit/', '');
            return <EmployeeEditForm employeeId={employeeId} />;
          }}
        />
        <Route component={NotFound} />
      </Switch>
    </div>
  </Router>
);

export default App;
