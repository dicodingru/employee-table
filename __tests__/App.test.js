import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/components/App';

describe('App', () => {
  test('should match snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
