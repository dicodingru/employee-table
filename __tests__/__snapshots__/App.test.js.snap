// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="container-fluid py-3"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="row"
      >
        <div
          className="col mb-3"
        >
          <button
            className="btn btn-primary d-block mx-auto ml-sm-auto mr-sm-0"
            data-target="#addModal"
            data-toggle="modal"
            type="button"
          >
            Добавить
          </button>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12 col-sm-3 col-md-2 mb-3"
        >
          <form
            className="form-inline d-md-block form-row"
          >
            <div
              className="col-6 col-sm mb-sm-3 text-sm-right"
            >
              <label
                className="sr-only"
                htmlFor="role"
              >
                Фильтровать по должности:
                 
              </label>
              <select
                className="form-control mw-100"
                id="role"
                onChange={[Function]}
                value="all"
              >
                <option
                  value="all"
                >
                  Должность
                </option>
                <option
                  value="driver"
                >
                  Водитель
                </option>
                <option
                  value="waiter"
                >
                  Официант
                </option>
                <option
                  value="cook"
                >
                  Повар
                </option>
              </select>
            </div>
            <div
              className="col-6 col-sm"
            >
              <div
                className="form-check d-block text-right"
              >
                <label
                  className="d-inline-block"
                  htmlFor="status"
                >
                  <input
                    checked={false}
                    className="form-check-input"
                    id="status"
                    onChange={[Function]}
                    type="checkbox"
                  />
                    В архиве
                </label>
              </div>
            </div>
          </form>
        </div>
        <div
          className="col-12 col-sm-9 col-md-10"
        >
          <div
            className="table-responsive"
          >
            <table
              className="table table-striped"
            >
              <thead
                className=""
              >
                <tr>
                  <th
                    className="bg-dark text-light"
                    onClick={[Function]}
                    scope="col"
                  >
                    ФИО
                    <i
                      aria-hidden="true"
                      className="fa float-right fa-sort-asc"
                    />
                  </th>
                  <th
                    scope="col"
                  >
                    Должность
                  </th>
                  <th
                    scope="col"
                  >
                    Телефон
                  </th>
                  <th
                    className=""
                    onClick={[Function]}
                    scope="col"
                  >
                    Дата рождения
                    <i
                      aria-hidden="true"
                      className="fa float-right fa-sort"
                    />
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <th
                    scope="row"
                  >
                    <a
                      href="/edit/7"
                      onClick={[Function]}
                    >
                      Александр Третьяков
                    </a>
                  </th>
                  <td>
                    Водитель
                  </td>
                  <td>
                    +7 (872) 568-2916
                  </td>
                  <td>
                    31.05.1979
                  </td>
                </tr>
                <tr>
                  <th
                    scope="row"
                  >
                    <a
                      href="/edit/5"
                      onClick={[Function]}
                    >
                      Алла Котова
                    </a>
                  </th>
                  <td>
                    Повар
                  </td>
                  <td>
                    +7 (948) 523-2964
                  </td>
                  <td>
                    26.01.1982
                  </td>
                </tr>
                <tr>
                  <th
                    scope="row"
                  >
                    <a
                      href="/edit/3"
                      onClick={[Function]}
                    >
                      Богдан Давыдов
                    </a>
                  </th>
                  <td>
                    Водитель
                  </td>
                  <td>
                    +7 (971) 575-2645
                  </td>
                  <td>
                    29.11.1990
                  </td>
                </tr>
                <tr>
                  <th
                    scope="row"
                  >
                    <a
                      href="/edit/11"
                      onClick={[Function]}
                    >
                      Валерий Пестов
                    </a>
                  </th>
                  <td>
                    Повар
                  </td>
                  <td>
                    +7 (899) 403-2387
                  </td>
                  <td>
                    20.01.1987
                  </td>
                </tr>
                <tr>
                  <th
                    scope="row"
                  >
                    <a
                      href="/edit/17"
                      onClick={[Function]}
                    >
                      Евдокия Филиппова
                    </a>
                  </th>
                  <td>
                    Официант
                  </td>
                  <td>
                    +7 (877) 450-3253
                  </td>
                  <td>
                    03.12.1994
                  </td>
                </tr>
                <tr>
                  <th
                    scope="row"
                  >
                    <a
                      href="/edit/1"
                      onClick={[Function]}
                    >
                      Илья Емельянов
                    </a>
                  </th>
                  <td>
                    Водитель
                  </td>
                  <td>
                    +7 (883) 508-3269
                  </td>
                  <td>
                    12.02.1982
                  </td>
                </tr>
                <tr>
                  <th
                    scope="row"
                  >
                    <a
                      href="/edit/8"
                      onClick={[Function]}
                    >
                      Пелагея Морозова
                    </a>
                  </th>
                  <td>
                    Водитель
                  </td>
                  <td>
                    +7 (977) 521-3479
                  </td>
                  <td>
                    11.09.1981
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <div
      className="modal"
      id="addModal"
    >
      <div
        className="modal-dialog"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-body"
          >
            <form
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="name"
                >
                  ФИО
                </label>
                <input
                  className="form-control"
                  id="name"
                  name="name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="phone"
                >
                  Телефон
                </label>
                <input
                  className="form-control"
                  id="phone"
                  name="phone"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="role"
                >
                  Должность
                </label>
                <select
                  className="form-control"
                  id="role"
                  name="role"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                    ---
                  </option>
                  <option
                    value="driver"
                  >
                    Водитель
                  </option>
                  <option
                    value="waiter"
                  >
                    Официант
                  </option>
                  <option
                    value="cook"
                  >
                    Повар
                  </option>
                </select>
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor=""
                >
                  Дата рождения
                </label>
                <input
                  className="form-control"
                  id="birthday"
                  name="birthday"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type="date"
                  value=""
                />
              </div>
              <div>
                <button
                  className="btn btn-primary d-block mx-auto"
                  type="submit"
                >
                  Добавить
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
